### DISCUSSION TITLE
umi新建项目并build后页面空白

### DISCUSSION THREAD

**Discussion by Shen-Shuyi** [umijs/umi#10612] [Q&A]
新建了umi项目, 且未添加任何自己的代码, start可以正常看到Page Index页面, 但build后双击index.html页面空白
一开始显示报错找不到umi.css和umi.js, 手动修改index.html将`/umi.css`,`/umi.js`改为`./umi.css`和`./umi.js`后报错消失且能正常下载到css和js文件, 但页面依然空白, 此时控制台没有任何报错, 网络也没有异常, 请问是什么原因?  
umi版本3.5.37
使用pnpm install下载依赖包

**Comment by fz6m** [comment]
需要给一个完整的复现视频，帮助排查问题。

**Reply by Shen-Shuyi** [reply]
https://www.bilibili.com/video/BV1wL411Z7Yi/?pop_share=1&vd_source=551a772e3f52daddc5baabb17fb554cf 这是我的录屏, 放在B站了

**Reply by fz6m** [reply]
umi 是一个 SPA 页面，打包完的产物需要用某种服务来提供，比如 nginx 等，在本地还可以用 serve (pnpm i -g serve) 来提供这些服务，直接打开是不可以的。

**Reply by Shen-Shuyi** [reply]
刷新页面可以了,谢谢

**Comment by qimuyunduan** [comment]
我的nginx 配置的根目录root是/var/www/www.maque.space/dist,umi build后，本地serve是可以的，上传dist后白屏。  
hash: true,
  history: {
    type: 'hash',
  },
  publicPath: './',
  base: '/',
以上是部分config配置

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4
